packages/react-component-library/src/forms/vanilla/vanilla.stories.tsx

```diff
@@ -0,0 +1,194 @@
+            onChange={handleCheckboxGroup}
+            name="exampleCheckbox"
+            label="Option 1"
+            value="Option 1"
+          />
+          <CheckboxE
+            onChange={handleCheckboxGroup}
+            name="exampleCheckbox"
+            label="Option 2"
+            value="Option 2"
+          />
+          <CheckboxE
+            onChange={handleCheckboxGroup}
+            name="exampleCheckbox"
+            label="Option 3"
+            value="Option 3"
+          />
+        </Fieldset>
+        <Fieldset>
+          <legend>Example radio selection</legend>
+          <RadioE
+            onChange={handleRadioGroup}
+            name="exampleRadio"
+            label="Option 1"
+            value="Option 1"
+          />
+          <RadioE
+            onChange={handleRadioGroup}
+            name="exampleRadio"
+            label="Option 2"
+            value="Option 2"
+          />
+        </Fieldset>
+        <SwitchE
+          name="exampleSwitch"
+          label="Example switch selection"
+          onChange={(e: React.FormEvent<HTMLInputElement>) =>
+            handleChange({
+              target: {
+                name: 'exampleSwitch',
+                value: e.currentTarget.value,
+              },
+            })
+          }
+          value={formState.exampleSwitch}
+          data-testid="form-example-SwitchE"
+        >
+          <SwitchEOption label="One" value="1" />
+          <SwitchEOption label="Two" value="2" />
+          <SwitchEOption label="Three" value="3" />
+        </SwitchE>
+        <NumberInputE
+          name="exampleNumberInput"
+          label="Example number input"
+          onChange={(
+            _:
+              | React.ChangeEvent<HTMLInputElement>
+              | React.MouseEvent<HTMLButtonElement>,
+            newValue: number
+          ) => {
+            handleChange({
+              target: {
+                name: 'exampleNumberInput',
+                value: newValue,
+              },
+            })
+          }}
+          value={formState.exampleNumberInput}
+          data-testid="form-example-NumberInputE"
+        />
+        <RangeSliderE
+          onChange={(values: ReadonlyArray<number>) => {
+            handleChange({
+              target: {
+                name: 'exampleRangeSlider',
+                value: values as [],
+              },
+            })
+          }}
+          values={formState.exampleRangeSlider}
+          domain={[0, 40]}
+          mode={1}
+          tracksLeft
+          step={2}
+        />
+        <ButtonE
+          type="submit"
+          data-testid="form-example-submit"
+          isDisabled={isSubmitting}
+        >
+          Submit
+        </ButtonE>
+      </form>
+
+      <pre data-testid="form-example-values">
+        {JSON.stringify(formPayload, null, 2)}
+      </pre>
+    </main>
+  )
+}
+
+export default {
+  title: 'Forms/Vanilla',
+  component: ExampleVanillaForm,
+} as Meta
```

Comment:
Btw they've changed the types a bit in Storybook 6.4:

![image](https://user-images.githubusercontent.com/66470099/148530494-6fb6917e-980c-46ec-853a-9bcb9f096d28.png)

But we need to update it everywhere...

Reply:
Changed for forms and created a [new ticket](https://github.com/defencedigital/mod-uk-design-system/issues/2961) to handle this across the rest of the codebase.